hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestHoodieDeltaStreamer.java

```diff
@@ -1908,19 +1932,15 @@ public void testPayloadClassUpdateWithCOWTable() throws Exception {
     assertTrue(props.containsKey(HoodieTableConfig.RECORD_MERGE_STRATEGY_ID.key()));
 
     //now create one more deltaStreamer instance and update payload class
-    cfg = TestHelpers.makeConfig(dataSetBasePath, WriteOperationType.BULK_INSERT,
+    HoodieDeltaStreamer.Config updatedConfig = TestHelpers.makeConfig(dataSetBasePath, WriteOperationType.BULK_INSERT,
         Collections.singletonList(SqlQueryBasedTransformer.class.getName()), PROPS_FILENAME_TEST_SOURCE, false,
         true, true, DummyAvroPayload.class.getName(), null);
-    new HoodieDeltaStreamer(cfg, jsc, fs, hiveServer.getHiveConf());
-
-    props = new Properties();
-    fs = HadoopFSUtils.getFs(cfg.targetBasePath, jsc.hadoopConfiguration());
-    try (InputStream inputStream = fs.open(new Path(metaPath))) {
-      props.load(inputStream);
-    }
 
-    //now using payload
-    assertEquals(DummyAvroPayload.class.getName(), props.get(HoodieTableConfig.PAYLOAD_CLASS_NAME.key()));
+    Exception e = assertThrows(HoodieException.class, () -> {
+      new HoodieDeltaStreamer(updatedConfig, jsc, fs, hiveServer.getHiveConf());
+    }, "Should error out when payload class is switched");
+    assertTrue(e.getMessage().contains("Config conflict(key"));
+    assertTrue(e.getMessage().contains(HoodieTableConfig.PAYLOAD_CLASS_NAME.key()));
```

Comment:
Could the full error message be asserted instead of part of the message which is confusing?